Gems/AtomLyIntegration/EMotionFXAtom/Code/Source/AtomActorDebugDraw.cpp

```diff
@@ -140,33 +214,66 @@ namespace AZ::Render
+                nodeDebugDrawDataFunction,
+                { renderActorSettings.m_hitDetectionColliderColor, renderActorSettings.m_selectedHitDetectionColliderColor });
         }
 
         // Cloth colliders
         if (CheckBitsAny(renderFlags, EMotionFX::ActorRenderFlags::ClothColliders))
         {
-            RenderColliders(debugDisplay, EMotionFX::PhysicsSetup::Cloth, instance,
-                renderActorSettings.m_clothColliderColor, renderActorSettings.m_selectedClothColliderColor);
+            m_characterPhysicsDebugDraw.RenderColliders(
+                debugDisplay, instance->GetActor()->GetPhysicsSetup()->GetColliderConfigByType(EMotionFX::PhysicsSetup::Cloth),
+                nodeDebugDrawDataFunction,
+                { renderActorSettings.m_clothColliderColor, renderActorSettings.m_selectedClothColliderColor });
         }
 
         // Simulated object colliders
         if (CheckBitsAny(renderFlags, EMotionFX::ActorRenderFlags::SimulatedObjectColliders))
         {
-            RenderColliders(debugDisplay, EMotionFX::PhysicsSetup::SimulatedObjectCollider, instance,
-                renderActorSettings.m_simulatedObjectColliderColor, renderActorSettings.m_selectedSimulatedObjectColliderColor);
+            m_characterPhysicsDebugDraw.RenderColliders(
+                debugDisplay,
+                instance->GetActor()->GetPhysicsSetup()->GetColliderConfigByType(EMotionFX::PhysicsSetup::SimulatedObjectCollider),
+                nodeDebugDrawDataFunction,
+                { renderActorSettings.m_simulatedObjectColliderColor, renderActorSettings.m_selectedSimulatedObjectColliderColor });
         }
 
         // Ragdoll
-        const bool renderRagdollColliders = AZ::RHI::CheckBitsAny(renderFlags, EMotionFX::ActorRenderFlags::RagdollColliders);
-        const bool renderRagdollJointLimits = AZ::RHI::CheckBitsAny(renderFlags, EMotionFX::ActorRenderFlags::RagdollJointLimits);
-        if (renderRagdollColliders || renderRagdollJointLimits)
+        if (AZ::RHI::CheckBitsAny(renderFlags, EMotionFX::ActorRenderFlags::RagdollColliders))
+        {
+            m_characterPhysicsDebugDraw.RenderColliders(
+                debugDisplay,
+                instance->GetActor()->GetPhysicsSetup()->GetColliderConfigByType(EMotionFX::PhysicsSetup::Ragdoll),
+                nodeDebugDrawDataFunction,
+                { renderActorSettings.m_ragdollColliderColor, renderActorSettings.m_selectedRagdollColliderColor });
+        }
+        if (AZ::RHI::CheckBitsAny(renderFlags, EMotionFX::ActorRenderFlags::RagdollJointLimits))
         {
-            RenderRagdoll(debugDisplay, instance, renderRagdollColliders, renderRagdollJointLimits);
+            m_characterPhysicsDebugDraw.RenderJointLimits(
+                debugDisplay, instance->GetActor()->GetPhysicsSetup()->GetRagdollConfig(), jointDebugDrawDataFunction,
+                { renderActorSettings.m_ragdollColliderColor, renderActorSettings.m_selectedRagdollColliderColor,
```

Comment:
nit: I've found from experience (even though its technically redundant) adding the struct/type name here can be super helpful when tracking down references, so I'd be in favour of adding it (could be a separate change if this has nearly cleared AR) but it made my life a lot simpler in a few cases and I regretted not doing it in a few others 🙈